### PR TITLE
add boxes to parsers

### PR DESCRIPTION
**Pull request by fgrunewald**
For multiple reasons it would be very useful, when the coordinate parsers would also store the box. I've added this parsing feature to the gro / pdb parser in the least invasive way. The gro solution is not so neat but then again the parser is also entirely out of line with the rest. 

For the PDB parser I stored the box in the parser object itself. In my opinion this would be safer and then we can decide when and if to use it with molecules. Unfortunately this doesn't work for the gro file reader so it then becomes a molecule class variable. I'm open to other suggestions, but would rather have it fast than redoing any major part of the parsers

### CODE REVIEW

File: vermouth/pdb/pdb.py [hunk 1]
```diff
@@ -265,6 +265,38 @@ def _atom(self, line, lineno=0):
...
+            ('beta', float, 7),
+            ('gamma', float, 7),
+            ('space_group', str, 11),
+            ('z_value', int, 4),
+            ]
+        start = 0
+        field_slices = []
+        for name, type_, width in fields:
+            if name == "space_group":
+                start+=1
```
Comment (pckroon) [inline vermouth/pdb/pdb.py:288]: lint.
Also, why this +1? If there's space in the format, take a look at how I solved it with the ATOM records. IIRC I put a negative width

File: vermouth/gmx/gro.py [hunk 2]
```diff
@@ -107,6 +107,11 @@ def read_gro(file_name, exclude=('SOL',), ignh=False):
 
             molecule.add_node(idx, **properties)
             idx += 1
+        box = np.array(line.strip().split(), dtype=float)
+        # not pretty but the parser is out of touch with the rest of the parsing
+        # infrastructure already
+        molecule.box = box
```
Comment (pckroon) [inline vermouth/gmx/gro.py:113]: What's the problem with this?

**Review by pckroon — CHANGES_REQUESTED** [review]
I like it. Tiny bit of lint and a small question/comment.
Also, could you hack these back into the writers as well?

File: vermouth/pdb/pdb.py (hunk 1 above)
Comment (fgrunewald) [reply vermouth/pdb/pdb.py:288]: yeah there is a gap in the fields between space_group and the next item.

File: vermouth/gmx/gro.py (hunk 2 above)
Comment (fgrunewald) [reply vermouth/gmx/gro.py:113]: Ideally I'd not like to add "secret" class variables to molecule. Makes it a bit hard to know when this attribute is set and what possible values are. Having said that the only other option would be to put it into the meta dict. Both things are possible. Also box should ideally propagate to system but that's perhaps something for when we deal with coordinate building etc.

Comment (fgrunewald) [reply vermouth/gmx/gro.py:113]: well we can go either way; if you like it we can also just add it to molecule and put a doctoring with it

**Comment by pckroon** [comment]
Keep it as is, and please open an issue complaining about the gro parser not returning a System; and to add something box-like to System

**Pull request merged**